File: 938B.py
n = int(input())
a = [1] + list(map(int, input().split())) + [pow(10, 6)]

ans = pow(10, 6)
l, r = 0, n + 1

while n > 0:
  if abs(a[l] - a[l + 1]) <= abs(a[r] - a[r - 1]):
    ans = min(ans, abs(a[l] - a[l + 1]))
    l += 1
  else:
    ans = min(ans, abs(a[r] - a[r - 1]))
    r -= 1

  n -= 1

print(ans)

File: 938B.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
  int n;
  cin >> n;

  vector<int> pos(n);

  for (int i = 0; i < n; i++) {
    cin >> pos[i];
  }

  int ans = 1e7;

  for (int i = 0; i < n; i++) {
    int curr = 0;
    if (i) curr = max(curr, pos[i - 1] - 1);
    if (i != n) curr = max(curr, 1000000 - pos[i]);

    ans = min(ans, curr);
  }

  cout << ans << endl;
}